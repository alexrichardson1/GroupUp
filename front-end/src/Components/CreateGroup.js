<<<<<<< HEAD
import React, { Component } from "react";
import axios from "axios";
import "./styles.css";
import { Form, Button, InputGroup, FormControl } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";

async function addGroup(data) {
  var result = {};
  await axios
    .post(`"http://localhost:5000/group/add`, data)
    .then((res) => {
      const group = res.data;
      console.log(group);
      result = group;
    })
    .catch((error) => {
      console.log(error);
    });
  return result;
}

export default class CreateGroup extends Component {
  onSubmit = async (event) => {
    event.preventDefault();
    const title = this.title.value;
    const tags = this.tags.value;
    const info = { title: title, tags: [tags] };
    await addGroup(info);
    window.location.reload();
  };
=======
import { Form, Button, InputGroup, FormControl } from "react-bootstrap";
import { useHistory } from "react-router-dom";

const CreateGroup = () => {
  let history = useHistory();

  return (
    <div>
      <h1>Create your group</h1>
      <Form>
        <Form.Group controlId="formLeaderName">
          <Form.Label>Leader's Full Name</Form.Label>
          <Form.Control type="text" placeholder="Leader's Full Name" />
          <Form.Text className="text-muted">
            This can really be any one of your names, it's just the name your
            group will be advertised under.
          </Form.Text>
        </Form.Group>
>>>>>>> 68c89e315e00af3e3342c1a8b45f61dc3feae2d4

  render() {
    return (
      <div>
        <h1>Create your group</h1>
        <Form>
          <Form.Group controlId="formLeaderName">
            <Form.Label>Leader's Full Name</Form.Label>
            <Form.Control type="text" placeholder="Leader's Full Name" />
            <Form.Text className="text-muted">
              This can really be any one of your names, it's just the name your
              group will be advertised under.
            </Form.Text>
          </Form.Group>

          <Form.Group controlId="formOtherMembers">
            <Form.Label>Other Members</Form.Label>
            <Form.Control type="text" placeholder="All other members go here" />
          </Form.Group>

<<<<<<< HEAD
          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="basic-addon1">Timezone</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="e.g. +5 = GMT+5"
              aria-label="Username"
              aria-describedby="basic-addon1"
            />
          </InputGroup>
=======
        <InputGroup className="mb-3">
          <InputGroup.Prepend>
            <InputGroup.Text id="basic-addon1">
              Programming Language
            </InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="e.g. Java"
            aria-label="Username"
            aria-describedby="basic-addon1"
          />
        </InputGroup>
>>>>>>> 68c89e315e00af3e3342c1a8b45f61dc3feae2d4

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="basic-addon1">
                Programming Language
              </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="e.g. Java"
              aria-label="Username"
              aria-describedby="basic-addon1"
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="basic-addon1">Nationality</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="e.g. United Kingdom"
              aria-label="Username"
              aria-describedby="basic-addon1"
            />
          </InputGroup>

<<<<<<< HEAD
          <Form.Group controlId="formOtherMembers">
            <Form.Label>Other Requirements</Form.Label>
            <Form.Control type="text" placeholder="" />
          </Form.Group>

          <Form.Group controlId="formMembersNeeded">
            <Form.Label>Members needed</Form.Label>
            <Form.Control as="select" custom>
              <option>1</option>
              <option>2</option>
              <option>3</option>
              <option>4</option>
              <option>5</option>
            </Form.Control>
          </Form.Group>
          <LinkContainer to="/listings">
            <Button>Go Back</Button>
          </LinkContainer>
          <Button variant="primary" type="submit" onSubmit={this.onSubmit}>
            Submit
          </Button>
        </Form>
      </div>
      // <div className="formWrap">
      //   <div className="formContent">
      //     <form className="form" onSubmit={this.onSubmit}>
      //       <h1 className="formH1">Enter Session Details</h1>
      //       <input
      //         type="text"
      //         className="formInput"
      //         placeholder="Title"
      //         ref={(input) => (this.title = input)}
      //       />
      //       <input
      //         type="text"
      //         className="formInput"
      //         placeholder="Tags"
      //         ref={(input) => (this.tags = input)}
      //       />
      //       <button className="createButton">Create Session</button>
      //     </form>
      //   </div>
      // </div>
    );
  }
}
=======
        <Form.Group controlId="formMembersNeeded">
          <Form.Label>Members needed</Form.Label>
          <Form.Control as="select" custom>
            <option>1</option>
            <option>2</option>
            <option>3</option>
            <option>4</option>
            <option>5</option>
          </Form.Control>
        </Form.Group>
        {/* <LinkContainer to="/listing//">
          <Button>Go Back</Button>
        </LinkContainer> */}
        <Button onClick={() => history.goBack()}>Go Back</Button>
        <Button variant="primary" type="submit">
          Submit
        </Button>
      </Form>
    </div>
  );
};

export default CreateGroup;
>>>>>>> 68c89e315e00af3e3342c1a8b45f61dc3feae2d4
