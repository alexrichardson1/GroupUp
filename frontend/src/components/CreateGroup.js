import React, { Component } from "react";
import axios from "axios";
import "components/styles.css";
import { Form, Button, InputGroup, FormControl } from "react-bootstrap";
import { withRouter } from "react-router-dom";
import NavBar from "components/NavBar";
// import { useHistory } from "react-router-dom";

async function addGroup(data) {
  var result = {};
  await axios
    .post(`http://localhost:5000/group/add`, data)
    .then((res) => {
      const group = res.data;
      result = group;
    })
    .catch((error) => {
      console.log(error);
    });
  return result;
}

class CreateGroup extends Component {
  onSubmit = async (event) => {
    event.preventDefault();
    const title = this.title.value;
    const tags = this.tags.value;
    const info = { title: title, tags: [tags] };
    this.goBack = this.goBack.bind(this);
    await addGroup(info);
    window.location.reload();
  };

  render() {
    //let history = useHistory();
    return (
      <div>
        <NavBar renderBool={[true, true, true, true]} create={true} />
        <h1>Create your group</h1>
        <Form>
          <Form.Group controlId="formLeaderName">
            <Form.Label>Leader's Full Name</Form.Label>
            <Form.Control type="text" placeholder="Leader's Full Name" />
            <Form.Text className="text-muted">
              This can really be any one of your names, it's just the name your
              group will be advertised under.
            </Form.Text>
          </Form.Group>

          <Form.Group controlId="formOtherMembers">
            <Form.Label>Other Members</Form.Label>
            <Form.Control type="text" placeholder="All other members go here" />
          </Form.Group>

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="basic-addon1">Timezone</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="e.g. +5 = GMT+5"
              aria-label="Username"
              aria-describedby="basic-addon1"
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="basic-addon1">
                Programming Language
              </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="e.g. Java"
              aria-label="Username"
              aria-describedby="basic-addon1"
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="basic-addon1">Nationality</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="e.g. United Kingdom"
              aria-label="Username"
              aria-describedby="basic-addon1"
            />
          </InputGroup>

          <Form.Group controlId="formOtherMembers">
            <Form.Label>Other Requirements</Form.Label>
            <Form.Control type="text" placeholder="" />
          </Form.Group>

          <Form.Group controlId="formMembersNeeded">
            <Form.Label>Members needed</Form.Label>
            <Form.Control as="select" custom>
              <option>1</option>
              <option>2</option>
              <option>3</option>
              <option>4</option>
              <option>5</option>
            </Form.Control>
          </Form.Group>
          {/* <LinkContainer to="/listing//">
          <Button>Go Back</Button>
        </LinkContainer> */}
          <Button onClick={() => this.props.history.goBack()}>Go Back</Button>
          <Button variant="primary" type="submit">
            Submit
          </Button>
        </Form>
      </div>
    );
  }
}

export default withRouter(CreateGroup);
