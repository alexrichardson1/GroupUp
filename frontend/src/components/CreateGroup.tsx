import { ChangeEvent, Component } from "react";
import axios from "axios";
import "components/styles.css";
import { config } from "Constants";
import {
  Form,
  Button,
  InputGroup,
  FormControl,
  Container,
  Col,
  Alert,
  Spinner,
} from "react-bootstrap";
import { withRouter } from "react-router-dom";
import { Redirect } from "react-router";
import NavBar from "components/NavBar";
import { UserContext } from "components/auth/UserContext";
import { ActiveT, GroupT, ProjectT } from "types/types";

async function addGroup(data: GroupT) {
  var result = {};
  await axios
    .post(`${config.API_URL}/group/add`, data)
    .then((res) => {
      const group = res.data;
      result = group;
    })
    .catch((error) => {
      console.log(error);
    });
  return result;
}

interface Props {
  id: number;
}

interface State {
  leader: string;
  maxmembers: number;
  teammates: string;
  requirements: string[];
  adrequirements: string;
  leaderemail: string;
  invalid: boolean;
  invalid2: boolean;
  redirect: boolean;
}

class CreateGroup extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      leader: "",
      leaderemail: "",
      maxmembers: 4,
      teammates: "",
      requirements: [],
      adrequirements: "",
      invalid: false,
      invalid2: false,
      redirect: false,
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  static override contextType = UserContext;

  getProject = async () => {
    var result: ProjectT = {
      id: 0,
      name: "",
      requirements: [],
      description: "",
      hours: 0,
      date: "",
      location: "",
    };
    await axios
      .post(`${config.API_URL}/project/one`, {
        projectid: this.props.id,
      })
      .then((res) => {
        const project = res.data;
        result = project;
      })
      .catch((error) => {
        console.log(error);
      });
    return result;
  };

  getActive = async () => {
    await axios
      .get(`${config.API_URL}/active/one`)
      .then((res) => {
        const user: ActiveT = res.data;
        this.setState({ leaderemail: user.email });
        // TODO
        //@ts-ignore
        this.setState({ leader: user.fullname });
        //@ts-ignore
        console.log(`EXPECT ERROR ${user.fullname}`);
      })
      .catch((error) => {
        console.error(error);
      });
  };

  override async componentDidMount() {
    this.getActive();
    const proj = await this.getProject();
    this.setState({ requirements: proj.requirements });
    document.title = "Create a listing!";
  }

  /* Functions to handle form submission */
  handleInputChange(event: ChangeEvent<HTMLInputElement>) {
    const target = event.target;
    const value = target.type === "checkbox" ? target.checked : target.value;

    //@ts-ignore
    this.setState({
      [target.name]: value,
    });
  }

  override componentWillUnmount() {
    //@ts-ignore
    clearTimeout(this.idss);
  }

  handleSubmit = async (e: any) => {
    let submit = true;
    this.state.requirements.forEach((req) => {
      // @ts-ignore
      if (this.state[req] === "" || this.state[req] === undefined) {
        submit = false;
      }
    });
    e.preventDefault();
    if (submit === true) {
      const requirementValues: string[] = [];
      this.state.requirements.map((req) =>
        //@ts-ignore
        requirementValues.push(this.state[req])
      );
      const info = {
        leader: this.state.leader,
        maxmembers: this.state.maxmembers,
        teammates: this.state.teammates.split(", "),
        requirements: requirementValues,
        adrequirements: this.state.adrequirements,
        projectid: this.props.id,
        leaderemail: this.state.leaderemail,
        posted: new Date().toISOString(),
      };
      console.log(info);
      await addGroup(info);
      this.setState({ invalid: false });
      const TIMEOUT = 3000;
      //@ts-ignore
      this.id = setTimeout(() => this.setState({ redirect: true }), TIMEOUT);
      this.setState({ invalid2: true });
    } else {
      this.setState({ invalid: true });
    }
  };

  uniqueComponent = (name: string) => {
    return (
      <InputGroup className="mb-3" key={name}>
        <InputGroup.Prepend>
          <InputGroup.Text id={name}>{name}</InputGroup.Text>
        </InputGroup.Prepend>
        <FormControl
          name={name}
          value={this.state.leader}
          // value={this.state[name]}
          onChange={this.handleInputChange}
          // isInvalid={!(this.state[name] && this.state[name].length !== 0)}
          isInvalid={!(this.state.leader && this.state.leader.length !== 0)}
        ></FormControl>
      </InputGroup>
    );
  };

  reqsToComponents = (names: string[]) => {
    const components: JSX.Element[] = [];
    names.forEach((req) => {
      components.push(this.uniqueComponent(req));
    });
    return components;
  };

  allFieldsNotFilledIn() {
    if (this.state.invalid) {
      return <Alert variant="danger">Fill in all fields.</Alert>
    }
    return null;
  }

  groupAdvertised() {
    if (this.state.invalid2) {
      return <Alert variant="success">Group advertised!</Alert>
    }
    return null;
  }

  override render() {
    return this.state.redirect ? (
      <Redirect to="/home" />
    ) : (
      <Container>
        <NavBar
          renderBool={[true, true, true, true]}
          create={true}
          id={this.props.id}
        />
        <h1>Advertise your group</h1>
        {this.allFieldsNotFilledIn()}
        {this.groupAdvertised()}
        {!this.state.invalid2 ? (
          <Form onSubmit={this.handleSubmit}>
            <Col>
              <Form.Group
                className="formBox"
                as={Col}
                controlId="formLeaderName"
                key="formLeaderName"
              >
                <Form.Label>Leader's Name</Form.Label>
                <Form.Control
                  disabled
                  type="text"
                  placeholder={this.state.leader}
                  name="leader"
                  value={this.state.leader}
                  // onChange={this.handleInputChange}
                />
                <Form.Text className="text-muted">
                  Leader is just the point of contact, it can really be any
                  group member.
                </Form.Text>
              </Form.Group>
              <Form.Group
                className="formBox"
                as={Col}
                controlId="formLeaderName"
                key="formLeaderName"
              >
                <Form.Label>Leader's Email</Form.Label>
                <Form.Control
                  disabled
                  type="text"
                  placeholder={this.state.leaderemail}
                  name="leaderemail"
                  value={this.state.leaderemail}
                />
                <Form.Text className="text-muted">
                  This is the first point of contact for new members, make sure
                  you are active on this email!
                </Form.Text>
              </Form.Group>
            </Col>

            <label htmlFor="basic-url" className="formBox" key="labelllll">
              A few things you need to specify about your group, to help
              understand the skills, interests and backgrounds your group
              possess:
            </label>

            {this.reqsToComponents(this.state.requirements).map((req) => {
              return req;
            })}

            <Form.Row>
              <Form.Group controlId="adrequirements" key="adrequirements">
                <Form.Label>Additional Notes</Form.Label>
                <Form.Control
                  type="text"
                  placeholder="e.g. Experience in Robotics"
                  name="adrequirements"
                  value={this.state.adrequirements}
                  onChange={this.handleInputChange}
                />
                <Form.Text className="text-muted">
                  Any of your own requirements.
                </Form.Text>
              </Form.Group>
            </Form.Row>

            <Form.Group
              className="formBox"
              controlId="formOtherMembers"
              key="formOtherMembers"
            >
              <Form.Label>Other Members names</Form.Label>
              <Form.Control
                type="text"
                placeholder="e.g. John, Mary, Adam"
                name="teammates"
                value={this.state.teammates}
                onChange={this.handleInputChange}
              />
              <Form.Text className="text-muted">
                Add Teammates here (comma-separated).
              </Form.Text>
            </Form.Group>

            <Button variant="primary" type="submit">
              Submit
            </Button>
          </Form>
        ) : (
          <Spinner animation="grow" />
        )}
      </Container>
    );
  }
}

//@ts-ignore
export default withRouter(CreateGroup);
