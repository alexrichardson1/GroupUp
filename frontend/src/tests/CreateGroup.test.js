import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import CreateGroup from "components/CreateGroup";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <CreateGroup />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
