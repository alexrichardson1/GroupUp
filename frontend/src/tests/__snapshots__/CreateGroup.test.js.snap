// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <nav
    className="navBar navbar navbar-expand-lg navbar-light fixed-top"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "color": "white",
        }
      }
    >
      <img
        alt="group up logo"
        className="logoImage"
        href="/home"
        onClick={[Function]}
        src="https://i.imgur.com/w92t1xH.png"
      />
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr=auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/home"
          href="/home"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Home
          <img
            alt="drop down arrow"
            className="processImage"
            src="/right-caret.svg"
          />
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/selection"
          href="/selection"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Select Hackathon
          <img
            alt="drop down arrow"
            className="processImage"
            src="/right-caret.svg"
          />
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/listing/undefined"
          href="/listing/undefined"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Group Listings
          <img
            alt="drop down arrow"
            className="processImage"
            src="/right-caret.svg"
          />
        </a>
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Create a group
        </span>
      </div>
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="ml-auto navHelp navbar-brand"
          href="/help"
          id="collasible-nav-dropdown"
          onClick={[Function]}
          title="Help"
        >
          <img
            alt="help"
            className="splitterImage"
            src="/info-circle.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <h1>
    Advertise your group
  </h1>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="col"
    >
      <div
        className="formBox form-group col"
      >
        <label
          className="form-label"
          htmlFor="formLeaderName"
        >
          Leader's Name
        </label>
        <input
          className="form-control"
          disabled={true}
          id="formLeaderName"
          name="leader"
          placeholder=""
          type="text"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          Leader is just the point of contact, it can really be any group member.
        </small>
      </div>
      <div
        className="formBox form-group col"
      >
        <label
          className="form-label"
          htmlFor="formLeaderName"
        >
          Leader's Email
        </label>
        <input
          className="form-control"
          disabled={true}
          id="formLeaderName"
          name="leaderemail"
          placeholder=""
          type="text"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          This is the first point of contact for new members, make sure you are active on this email!
        </small>
      </div>
    </div>
    <label
      className="formBox"
      htmlFor="basic-url"
    >
      A few things you need to specify about your group, to help understand the skills, interests and backgrounds your group possess:
    </label>
    <div
      className="form-row"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="adrequirements"
        >
          Additional Notes
        </label>
        <input
          className="form-control"
          id="adrequirements"
          name="adrequirements"
          onChange={[Function]}
          placeholder="e.g. Experience in Robotics"
          type="text"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          Any of your own requirements.
        </small>
      </div>
    </div>
    <div
      className="formBox form-group"
    >
      <label
        className="form-label"
        htmlFor="formOtherMembers"
      >
        Other Members names
      </label>
      <input
        className="form-control"
        id="formOtherMembers"
        name="teammates"
        onChange={[Function]}
        placeholder="e.g. John, Mary, Adam"
        type="text"
        value=""
      />
      <small
        className="text-muted form-text"
      >
        Add Teammates here (comma-separated).
      </small>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
